python/cudf/cudf/core/frame.py

```diff
@@ -1761,11 +1762,16 @@ def sample(
                     "weights is not yet supported for axis=0/index"
                 )
 
-            seed = (
-                np.random.randint(np.iinfo(np.int64).max, dtype=np.int64)
-                if random_state is None
-                else np.int64(random_state)
-            )
+            if random_state is None:
+                seed = np.random.randint(
+                    np.iinfo(np.int64).max, dtype=np.int64
+                )
+            elif isinstance(random_state, np.random.mtrand.RandomState):
+                pos = random_state._bit_generator.state["state"]["pos"]
+                keys = random_state._bit_generator.state["state"]["key"]
```

Comment:
Reaching into `_bit_generator` feels very fragile. Could we do anything that doesn't require reaching into internal attributes?

Reply:
We can use get_states, but it is just a list, so we might have to index it with integers